### ISSUE TITLE
[BUG] 3 Node AirGapped Cluster Upgrade Stuck v1.1.0->v1.1.2-rc4

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#3730]
**Describe the bug**
The upgrade seems to be stuck at 1 Node Succeeding, 1 Node Predrained, 1 Node Images Preloaded

**Node About Hardware**
- The hardware that provides the qemu/kvm has been behaving strangely, occasionally on the iLO it seems to be having issues with the system disk controller.
- I will be trying to reproduce, but in the meantime setting to "reproduce/rare"

```
  System information as of Tue Mar 28 05:11:00 PM UTC 2023

  System load:  15.48291015625      Users logged in:         1
  Usage of /:   60.1% of 801.14GB   IPv4 address for eno49:  172.XX.XX.XXX
  Memory usage: 41%                 IPv4 address for eno50:  172.XX.XX.XXX
  Swap usage:   47%                 IPv4 address for virbr0: 192.168.XXX.X
  Temperature:  62.0 C              IPv4 address for virbr1: 192.168.XXX.X
  Processes:    509                 IPv4 address for virbr2: 192.168.XXX.X
```

**To Reproduce**
1. Have an airgapped 3 Node cluster v1.1.0->v1.1.2-rc4
1. Preform an upgrade after changing the systemUpgradeActiveDeadlineSeconds on the deployment for the upgrade controller

**Expected behavior**
The upgrade to not be stuck

**Support bundle**
[supportbundle_783291ed-7823-414c-9797-2bb412991adc_2023-03-28T17-10-28Z.zip](https://github.com/harvester/harvester/files/11092570/supportbundle_783291ed-7823-414c-9797-2bb412991adc_2023-03-28T17-10-28Z.zip)

**Environment**
 - Harvester ISO version: v1.1.0
 - Underlying Infrastructure: qemu/kvm

**Additional context**
![Screenshot from 2023-03-28 10-14-32](https://user-images.githubusercontent.com/5370752/228321529-899f275e-8e3b-4564-af6c-846b171a0c0b.png)
![Screenshot from 2023-03-28 10-13-45](https://user-images.githubusercontent.com/5370752/228321533-5e708c18-eecc-46c5-948a-8b455d6706fb.png)

**journalctl from harvester-node-0**
```
ubuntu@ubuntu-v110-rel-testing-airgap:~$ ssh rancher@192.168.2.30
(rancher@192.168.2.30) Password: 
Last login: Tue Mar 28 17:16:09 2023 from 192.168.2.1
rancher@harvester-node-0:~> sudo journalctl -p 0..3
Mar 24 21:06:59 harvester-node-0 rpcbind[502]: cannot open file = /run/rpcbind/rpcbind.xdr for writing
Mar 24 21:06:59 harvester-node-0 rpcbind[502]: cannot save any registration
Mar 24 21:06:59 harvester-node-0 rpcbind[502]: cannot open file = /run/rpcbind/portmap.xdr for writing
Mar 24 21:06:59 harvester-node-0 rpcbind[502]: cannot save any registration
Mar 24 21:45:52 harvester-node-0 sshd[26415]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 24 21:46:39 harvester-node-0 kernel: blk_update_request: critical medium error, dev sda, sector 50602632 op 0x1:(WRITE) flags 0x800 phys_seg 4 prio class 0
Mar 24 21:46:39 harvester-node-0 kernel: Aborting journal on device sda-8.
Mar 24 21:46:39 harvester-node-0 kernel: blk_update_request: critical medium error, dev sda, sector 50593792 op 0x1:(WRITE) flags 0x20800 phys_seg 1 prio class 0
Mar 24 21:46:39 harvester-node-0 kernel: Buffer I/O error on dev sda, logical block 6324224, lost sync page write
Mar 24 21:46:39 harvester-node-0 kernel: JBD2: Error -5 detected when updating journal superblock for sda-8.
Mar 24 21:46:39 harvester-node-0 kernel: blk_update_request: critical medium error, dev sda, sector 0 op 0x1:(WRITE) flags 0x20800 phys_seg 1 prio class 0
Mar 24 21:46:39 harvester-node-0 kernel: Buffer I/O error on dev sda, logical block 0, lost sync page write
Mar 24 21:46:39 harvester-node-0 kernel: EXT4-fs (sda): I/O error while writing superblock
Mar 24 21:46:39 harvester-node-0 kernel: EXT4-fs error (device sda): ext4_journal_check_start:61: Detected aborted journal
Mar 24 21:46:39 harvester-node-0 kernel: EXT4-fs (sda): Remounting filesystem read-only
Mar 24 21:46:39 harvester-node-0 kernel: blk_update_request: critical medium error, dev sda, sector 0 op 0x1:(WRITE) flags 0x20800 phys_seg 1 prio class 0
Mar 24 21:46:39 harvester-node-0 kernel: Buffer I/O error on dev sda, logical block 0, lost sync page write
Mar 24 21:46:39 harvester-node-0 kernel: EXT4-fs (sda): I/O error while writing superblock
Mar 24 21:47:16 harvester-node-0 kernel: blk_update_request: critical medium error, dev sdb, sector 4456816 op 0x1:(WRITE) flags 0x800 phys_seg 3 prio class 0
Mar 24 21:47:16 harvester-node-0 kernel: Aborting journal on device sdb-8.
Mar 24 21:47:16 harvester-node-0 kernel: blk_update_request: critical medium error, dev sdb, sector 4456448 op 0x1:(WRITE) flags 0x20800 phys_seg 1 prio class 0
Mar 24 21:47:16 harvester-node-0 kernel: Buffer I/O error on dev sdb, logical block 557056, lost sync page write
Mar 24 21:47:16 harvester-node-0 kernel: JBD2: Error -5 detected when updating journal superblock for sdb-8.
Mar 24 21:47:22 harvester-node-0 kernel: blk_update_request: I/O error, dev sdc, sector 0 op 0x1:(WRITE) flags 0x800 phys_seg 0 prio class 0
Mar 24 21:47:25 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:25 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:35 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:35 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:38 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:38 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:45 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:45 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:48 harvester-node-0 kernel: blk_update_request: I/O error, dev sdb, sector 0 op 0x1:(WRITE) flags 0x800 phys_seg 0 prio class 0
Mar 24 21:47:55 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:47:55 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:05 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:05 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:08 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:08 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:15 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:15 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:25 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:25 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:25 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_get_inode_loc:4744: inode #2: block 289: comm longhorn-manage: unable to read itable b>
Mar 24 21:48:25 harvester-node-0 kernel: EXT4-fs error (device sdc) in ext4_reserve_inode_write:5997: IO failure
Mar 24 21:48:31 harvester-node-0 kernel: Aborting journal on device sdc-8.
Mar 24 21:48:31 harvester-node-0 kernel: Buffer I/O error on dev sdc, logical block 262144, lost sync page write
Mar 24 21:48:31 harvester-node-0 kernel: JBD2: Error -5 detected when updating journal superblock for sdc-8.
Mar 24 21:48:35 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:35 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:38 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:38 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:45 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:45 harvester-node-0 kernel: EXT4-fs error (device sdc): __ext4_find_entry:1578: inode #2: comm grafana-server: reading directory lblock 0
Mar 24 21:48:45 harvester-node-0 kernel: EXT4-fs error (device sdc): ext4_wait_block_bitmap:519: comm grafana-server: Cannot read block bitmap - block_group = 1, >
Mar 24 21:48:45 harvester-node-0 kernel: EXT4-fs error (device sdc): ext4_discard_preallocations:4104: comm grafana-server: Error -5 reading block bitmap for 1
Mar 24 21:48:46 harvester-node-0 kernel: EXT4-fs error (device sdb): __ext4_find_entry:1578: inode #131073: comm alertmanager: reading directory lblock 0
Mar 24 21:48:46 harvester-node-0 kernel: EXT4-fs error (device sdb): __ext4_find_entry:1578: inode #131073: comm alertmanager: reading directory lblock 0
Mar 24 21:48:48 harvester-node-0 kernel: EXT4-fs error (device sda): ext4_wait_block_bitmap:519: comm umount: Cannot read block bitmap - block_group = 1, block_bi>
Mar 24 21:48:48 harvester-node-0 kernel: EXT4-fs error (device sda): ext4_discard_preallocations:4104: comm umount: Error -5 reading block bitmap for 1
Mar 24 21:54:49 harvester-node-0 kernel: blk_update_request: I/O error, dev sdd, sector 0 op 0x1:(WRITE) flags 0x800 phys_seg 0 prio class 0
-- Boot 191deddd09094d42a7b16e506c8d7dcb --
Mar 27 18:45:21 harvester-node-0 rpcbind[503]: cannot open file = /run/rpcbind/rpcbind.xdr for writing
Mar 27 18:45:21 harvester-node-0 rpcbind[503]: cannot save any registration
Mar 27 18:45:21 harvester-node-0 rpcbind[503]: cannot open file = /run/rpcbind/portmap.xdr for writing
Mar 27 18:45:21 harvester-node-0 rpcbind[503]: cannot save any registration
Mar 27 19:46:54 harvester-node-0 sshd[25107]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 27 20:08:19 harvester-node-0 sshd[25038]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 27 20:28:52 harvester-node-0 sshd[23109]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 27 20:34:57 harvester-node-0 sshd[32410]: error: PAM: User not known to the underlying authentication module for illegal user vagrant from 192.168.2.1
Mar 27 20:35:00 harvester-node-0 sshd[32410]: error: PAM: User not known to the underlying authentication module for illegal user vagrant from 192.168.2.1
Mar 27 20:37:15 harvester-node-0 sshd[3899]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 27 22:09:00 harvester-node-0 systemd[1]: Failed to start Rancher Kubernetes Engine v2 (server).
Mar 27 22:17:10 harvester-node-0 wickedd[1466]: __ni_rtnl_send_delroute(ipv4 0.0.0.0/0 via 192.168.2.254 dev mgmt-br#4 type unicast table main scope universe prot>
-- Boot a36d6742d25b44b081f50b7bf2e60b20 --
Mar 27 22:21:05 harvester-node-0 wickedd-nanny[1590]: /org/opensuse/Network/Interface/8.getManagedObjects failed. Server responds:
Mar 27 22:21:05 harvester-node-0 wickedd-nanny[1590]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.fre>
Mar 27 22:21:05 harvester-node-0 wickedd-nanny[1590]: /org/opensuse/Network/Interface/9.getManagedObjects failed. Server responds:
Mar 27 22:21:05 harvester-node-0 wickedd-nanny[1590]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.fre>
Mar 28 17:16:09 harvester-node-0 sshd[18101]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 28 17:22:43 harvester-node-0 sshd[28657]: pam_systemd(sshd:session): Acquired user record does not have a UID.
```

**journalctl from harvester node 1**
```
ubuntu@ubuntu-v110-rel-testing-airgap:~$ ssh rancher@192.168.2.31
(rancher@192.168.2.31) Password: 
Last login: Tue Mar 28 17:16:58 2023 from 192.168.2.1
rancher@harvester-node-1:~> sudo journalctl -p 3 -xb
-- Logs begin at Fri 2023-03-24 21:28:34 UTC, end at Tue 2023-03-28 17:19:49 UTC. --
Mar 27 18:45:31 harvester-node-1 rpcbind[503]: cannot open file = /run/rpcbind/rpcbind.xdr for writing
Mar 27 18:45:31 harvester-node-1 rpcbind[503]: cannot save any registration
Mar 27 18:45:31 harvester-node-1 rpcbind[503]: cannot open file = /run/rpcbind/portmap.xdr for writing
Mar 27 18:45:31 harvester-node-1 rpcbind[503]: cannot save any registration
Mar 27 22:28:53 harvester-node-1 wickedd-nanny[1484]: /org/opensuse/Network/Interface/74.getManagedObjects failed. Server responds:
Mar 27 22:28:53 harvester-node-1 wickedd-nanny[1484]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.fre>
Mar 28 17:16:58 harvester-node-1 sshd[29947]: pam_systemd(sshd:session): Acquired user record does not have a UID.
Mar 28 17:19:42 harvester-node-1 sshd[2064]: pam_systemd(sshd:session): Acquired user record does not have a UID.
```

**journalctl from harvester node 2**
```
rancher@harvester-node-2:~> sudo journalctl -p 3
-- Logs begin at Fri 2023-03-24 21:43:22 UTC, end at Tue 2023-03-28 17:18:54 UTC. --
Mar 24 21:43:29 harvester-node-2 rpcbind[498]: cannot open file = /run/rpcbind/rpcbind.xdr for writing
Mar 24 21:43:29 harvester-node-2 rpcbind[498]: cannot save any registration
Mar 24 21:43:29 harvester-node-2 rpcbind[498]: cannot open file = /run/rpcbind/portmap.xdr for writing
Mar 24 21:43:29 harvester-node-2 rpcbind[498]: cannot save any registration
Mar 24 21:45:50 harvester-node-2 sshd[9673]: pam_systemd(sshd:session): Acquired user record does not have a UID.
-- Reboot --
Mar 27 18:45:40 harvester-node-2 rpcbind[501]: cannot open file = /run/rpcbind/rpcbind.xdr for writing
Mar 27 18:45:40 harvester-node-2 rpcbind[501]: cannot save any registration
Mar 27 18:45:40 harvester-node-2 rpcbind[501]: cannot open file = /run/rpcbind/portmap.xdr for writing
Mar 27 18:45:40 harvester-node-2 rpcbind[501]: cannot save any registration
Mar 27 18:47:25 harvester-node-2 wickedd-nanny[1478]: /org/opensuse/Network/Interface/16.getManagedObjects failed. Server responds:
Mar 27 18:47:25 harvester-node-2 wickedd-nanny[1478]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.fre>
Mar 27 22:29:07 harvester-node-2 wickedd-nanny[1478]: /org/opensuse/Network/Interface/90.getManagedObjects failed. Server responds:
Mar 27 22:29:07 harvester-node-2 wickedd-nanny[1478]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.fre>
Mar 28 17:17:37 harvester-node-2 sshd[14762]: pam_systemd(sshd:session): Acquired user record does not have a UID.
```

**Comment by Vicente-Cheng** [comment]
Analyst this case here:
The PDB instance-manager-r-466e3c7f is created with the wrong node label (it should be harvester-node-2, but now harvester-node-1). So now stuck on draining. refer the rancher logs
```
error when evicting pods/"instance-manager-r-d4ed2788" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
```
harvester-node-1 log:
```
2023-03-24T21:46:16.326194608Z stderr F time="2023-03-24T21:46:16Z" level=info msg="Created instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-1
.
.
.
2023-03-24T21:47:44.816905647Z stderr F time="2023-03-24T21:47:44Z" level=info msg="Created instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-1
2023-03-24T21:47:44.82323971Z stderr F time="2023-03-24T21:47:44Z" level=debug msg="Start monitoring instance manager instance-manager-r-466e3c7f" controller=longhorn-instance-manager instance manager=instance-manager-r-466e3c7f node=harvester-node-1
.
.
.
// stopped, but wrongly PDB is still here.
2023-03-24T21:48:07.825269887Z stderr F time="2023-03-24T21:48:07Z" level=info msg="stop monitoring the instance manager on this node (harvester-node-1) because the instance manager has new ownerID harvester-node-2" controller=longhorn-instance-manager instance manager=instance-manager-r-466e3c7f node=harvester-node-1
2023-03-24T21:48:07.825431484Z stderr F time="2023-03-24T21:48:07Z" level=debug msg="Stop monitoring instance manager instance-manager-r-466e3c7f" controller=longhorn-instance-manager instance manager=instance-manager-r-466e3c7f node=harvester-node-1
2023-03-24T21:48:07.825450463Z stderr F time="2023-03-24T21:48:07Z" level=debug msg="removed the instance manager from imc.instanceManagerMonitorMap" controller=longhorn-instance-manager instance manager=instance-manager-r-466e3c7f node=harvester-node-1
2023-03-24T21:48:07.825462864Z stderr F time="2023-03-24T21:48:07Z" level=error msg="error receiving next item in engine watch: rpc error: code = Canceled desc = context canceled" controller=longhorn-instance-manager instance manager=instance-manager-r-466e3c7f node=harvester-node-1
```
harvester-node-2 log
```
2023-03-24T21:44:48.160841713Z stderr F time="2023-03-24T21:44:48Z" level=debug msg="Prepare to create default instance manager instance-manager-r-466e3c7f, node: harvester-node-2, default instance manager image: longhornio/longhorn-instance-manager:v1_20221003, type: replica"
2023-03-24T21:44:48.179369823Z stderr F time="2023-03-24T21:44:48Z" level=debug msg="Instance Manager Controller harvester-node-2 picked up instance-manager-r-466e3c7f" controller=longhorn-instance-manager instanceManager=instance-manager-r-466e3c7f node=harvester-node-2 nodeID=harvester-node-2
2023-03-24T21:44:48.179395489Z stderr F time="2023-03-24T21:44:48Z" level=info msg="Instance manager state is updated from  to stopped after syncing with the pod" controller=longhorn-instance-manager instanceManager=instance-manager-r-466e3c7f node=harvester-node-2 nodeID=harvester-node-2
2023-03-24T21:44:48.20987553Z stderr F time="2023-03-24T21:44:48Z" level=info msg="Created instance manager pod" controller=longhorn-instance-manager instanceManager=instance-manager-r-466e3c7f node=harvester-node-2 nodeID=harvester-node-2
.
.
.
2023-03-24T21:45:07.440440853Z stderr F time="2023-03-24T21:45:07Z" level=info msg="Instance manager state is updated from starting to running after syncing with the pod" controller=longhorn-instance-manager instanceManager=instance-manager-r-466e3c7f node=harvester-node-2 nodeID=harvester-node-2
2023-03-24T21:45:07.454588213Z stderr F time="2023-03-24T21:45:07Z" level=info msg="Created instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-2
2023-03-24T21:45:07.465901278Z stderr F time="2023-03-24T21:45:07Z" level=debug msg="Start monitoring instance manager instance-manager-r-466e3c7f" controller=longhorn-instance-manager instance manager=instance-manager-r-466e3c7f node=harvester-node-2
.
.
.
2023-03-24T21:46:13.75776607Z stderr F time="2023-03-24T21:46:13Z" level=debug msg="Node harvester-node-2 is marked unschedulable, try to remove instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-2
2023-03-24T21:46:13.81315013Z stderr F time="2023-03-24T21:46:13Z" level=info msg="Deleted instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-2
.
.
.
2023-03-24T21:46:16.353317832Z stderr F time="2023-03-24T21:46:16Z" level=debug msg="Node harvester-node-2 is marked unschedulable, try to remove instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-2
2023-03-24T21:46:16.406748066Z stderr F time="2023-03-24T21:46:16Z" level=info msg="Deleted instance-manager-r-466e3c7f PDB" controller=longhorn-instance-manager node=harvester-node-2
.
.
.
// get some error with logs
2023-03-27T18:48:26.5972967Z stderr F time="2023-03-27T18:48:26Z" level=warning msg="Dropping Longhorn replica longhorn-system/pvc-8c95a1a5-472c-421c-906a-155e30e964ef-r-0df56306 out of the queue" ... Status:{InstanceStatus:{OwnerID:harvester-node-2 InstanceManagerName:instance-manager-r-466e3c7f CurrentState:running CurrentImage:longhornio/longhorn-engine:v1.3.2 IP:10.52.2.14 StorageIP: Port:10000 Started:true LogFetched:false SalvageExecuted:false} EvictionRequested:false}}" node=harvester-node-2
.
.
.
2023-03-27T18:48:26.614377788Z stderr F time="2023-03-27T18:48:26Z" level=warning msg="number of node disks 1 and collected disk info 0 are not equal"
2023-03-27T18:48:26.614389621Z stderr F time="2023-03-27T18:48:26Z" level=info msg="mismatching disks in node resource object and monitor collected data"
2023-03-27T18:48:26.672185806Z stderr F time="2023-03-27T18:48:26Z" level=warning msg="Deleted instance manager pod instance-manager-r-466e3c7f for instance manager instance-manager-r-466e3c7f"
2023-03-27T18:48:26.69852629Z stderr F time="2023-03-27T18:48:26Z" level=warning msg="Deleted instance manager pod instance-manager-e-3da52100 for instance manager instance-manager-e-3da52100"
.
.
.
// recreate
2023-03-27T18:48:26.815702097Z stderr F time="2023-03-27T18:48:26Z" level=info msg="Created instance manager pod" controller=longhorn-instance-manager instanceManager=instance-manager-r-466e3c7f node=harvester-node-2 nodeID=harvester-node-2
2023-03-27T18:48:30.497613621Z stderr F time="2023-03-27T18:48:30Z" level=info msg="Instance manager state is updated from starting to running after syncing with the pod" controller=longhorn-instance-manager instanceManager=instance-manager-r-466e3c7f node=harvester-node-2 nodeID=harvester-node-2
```

The PDB (`instance-manager-r-466e3c7f`) life cycle is : PDB create (harvester-node-2) → PDB delete (harvester-node-2) → PDB create (harvester-node-1) → PDB delete (harvester-node-2) → PDB create (harvester-node-1)

The PDB (`instance-manager-r-466e3c7f`) info:
```
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  creationTimestamp: "2023-03-24T21:47:44Z"
  generation: 1
  name: instance-manager-r-466e3c7f
  namespace: longhorn-system
  resourceVersion: "3348508"
  uid: 54ffdc56-171a-4812-81a5-7404046da0df
spec:
  minAvailable: 1
  selector:
    matchLabels:
      longhorn.io/component: instance-manager
      longhorn.io/instance-manager-image: imi-97ee1183
      longhorn.io/instance-manager-type: replica
      longhorn.io/managed-by: longhorn-manager
      longhorn.io/node: harvester-node-1
status:
  conditions:
  - lastTransitionTime: "2023-03-27T22:13:28Z"
    message: ""
    observedGeneration: 1
    reason: InsufficientPods
    status: "False"
    type: DisruptionAllowed
  currentHealthy: 1
  desiredHealthy: 1
  disruptionsAllowed: 0
  expectedPods: 1
  observedGeneration: 1
```

**Comment by Vicente-Cheng** [comment]
related Longhorn issue: https://github.com/longhorn/longhorn/issues/5680

**Comment by bk201** [comment]
We also need the fix in 1.1.3; we need to bump the Longhorn version to v1.3.3.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.2.0-rc3`. 

### Result
<details><summary>The PDB instance-manager-r can be recreated with the correct node label  (change the lable) $~~$ $\color{green}{\textsf{PASS}}$</summary>

  * When we set taint on the node1 and delete the longhorn-manager pod of node1 
  * After we delete the related PDB instance manger eg. `im-r` and `im-e`
  
  1. The PDB content of `im-e` PDB, can be recreated on node1
        ```
        spec:                                                                                                                                                                                      │
        │   minAvailable: 1                                                                                                                                                                          │
        │   selector:                                                                                                                                                                                │
        │     matchLabels:                                                                                                                                                                           │
        │       longhorn.io/component: instance-manager                                                                                                                                              │
        │       longhorn.io/instance-manager-image: imi-c428551c                                                                                                                                     │
        │       longhorn.io/instance-manager-type: engine                                                                                                                                            │
        │       longhorn.io/managed-by: longhorn-manager                                                                                                                                             │
        │       longhorn.io/node: node1
        ```
  1. The PDB content of `im-r` PDB, can be recreated on node1
        ```
         spec:                                                                                                                                                                                      │
        │   minAvailable: 1                                                                                                                                                                          │
        │   selector:                                                                                                                                                                                │
        │     matchLabels:                                                                                                                                                                           │
        │       longhorn.io/component: instance-manager                                                                                                                                              │
        │       longhorn.io/instance-manager-image: imi-c428551c                                                                                                                                     │
        │       longhorn.io/instance-manager-type: replica                                                                                                                                           │
        │       longhorn.io/managed-by: longhorn-manager                                                                                                                                             │
        │       longhorn.io/node: node1
        ```

 </details> 

### Test Information
* Test Environment: 3 nodes harvester on bare machine
* Harvester version: `v1.2.0-rc3` 

### Verify Steps

<details><summary>Test steps to recreate pdb instance manager</summary>

1. Create a fresh new or upgrade a 3-node cluster to latest v1.2.0-rc3
2. Launch K9s on Harvester management node
3. Check the longhorn-manger version is `v1.4.2`
  ![image](https://github.com/harvester/harvester/assets/29251855/1c96a62e-7d29-450b-989c-335f4eadee1e)

4. Check the the instance manager of node1 is `e.g longhorn-manager-gmzvh` 
  ![image](https://github.com/harvester/harvester/assets/29251855/119d5eed-c516-4b50-ad51-752d629a4ac9)

5. Check the related PDB resource (instance-manger-r) on node1 (e.g instance-manager-r-1e8d2b6ac4bdab53558aa36fa56425b5) and it's point to node1

6. Add the taint `test=:NoSchedule` to node1 (refer to https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)

  ```
  node3:~ # kubectl taint nodes node1 key1=test:NoSchedule
  node/node1 tainted
  ```

8. Delete longhorn-manager-gmzvh pod on node1 (It could not recreated)
9.  Delete `instance-manager-r-xxx` PDB manually and observer that im-r PDB will recreate.
10. Delete `instance-manager-e-xxx` PDB manually and observer that im-e PDB will recreate.
  ![image](https://github.com/harvester/harvester/assets/29251855/6a1ca3a3-ef78-4ede-9932-6ee95f4426f6)

11. Check the PDB content of `im-e` PDB, and they should be create on node1
    ```
    spec:                                                                                                                                                                                      │
    │   minAvailable: 1                                                                                                                                                                          │
    │   selector:                                                                                                                                                                                │
    │     matchLabels:                                                                                                                                                                           │
    │       longhorn.io/component: instance-manager                                                                                                                                              │
    │       longhorn.io/instance-manager-image: imi-c428551c                                                                                                                                     │
    │       longhorn.io/instance-manager-type: engine                                                                                                                                            │
    │       longhorn.io/managed-by: longhorn-manager                                                                                                                                             │
    │       longhorn.io/node: node1
    ```
12. Check the PDB content of `im-r` PDB, and they should be create on node1
    ```
     spec:                                                                                                                                                                                      │
    │   minAvailable: 1                                                                                                                                                                          │
    │   selector:                                                                                                                                                                                │
    │     matchLabels:                                                                                                                                                                           │
    │       longhorn.io/component: instance-manager                                                                                                                                              │
    │       longhorn.io/instance-manager-image: imi-c428551c                                                                                                                                     │
    │       longhorn.io/instance-manager-type: replica                                                                                                                                           │
    │       longhorn.io/managed-by: longhorn-manager                                                                                                                                             │
    │       longhorn.io/node: node1
    ```

 </details>

**Comment by TachunLin** [comment]
@irishgordo, we have fixed and verified the root cause of upgrading stuck on 3 nodes airgapped cluster as stated in the description. 

I think it would be better we can double confirm this fix can make the airgapped cluster smoothly upgrading from v1.1.2 to v1.2.0 on your environment. 

If possible could you please also check whether this issue did not happen during the 3 nodes upgrade in the upcoming release testing. Many thanks.

**Comment by irishgordo** [comment]
@TachunLin - as I tried to test this w/ v1.2.0-rc4 I'm encountering an issue with the nodes being stuck at `images preloaded` phase, I opened up:
https://github.com/harvester/harvester/issues/4288

**Comment by irishgordo** [comment]
@TachunLin I tested this again and everything looks good - the problem was in the setup the need to adjust both the disk size on the nodes & the libvirt loader
![Screenshot from 2023-07-24 08-39-37](https://github.com/harvester/harvester/assets/5370752/2259c21b-42d4-4115-93fa-169411023659)
but this looks good with rc4 as well :smile: :+1:  - so I will go ahead and close this out

And also closed out #4288 :smile:
